exchanges/orderbook/calculator.go

```diff
@@ -15,214 +18,186 @@ type WhaleBombResult struct {
-
-func (o orderSummary) Print() {
-	for x := range o {
-		log.Debugf(log.OrderBook, "Order: Price: %f Amount: %f", o[x].Price, o[x].Amount)
-	}
-}
-
-func (o orderSummary) MinimumPrice(reverse bool) float64 {
-	if len(o) != 0 {
-		sortOrdersByPrice(&o, reverse)
-		return o[0].Price
-	}
-	return 0
-}
-
-func (o orderSummary) MaximumPrice(reverse bool) float64 {
-	if len(o) != 0 {
-		sortOrdersByPrice(&o, reverse)
-		return o[0].Price
-	}
-	return 0
-}
-
-// ByPrice used for sorting orders by order date
-type ByPrice orderSummary
-
-func (b ByPrice) Len() int           { return len(b) }
-func (b ByPrice) Less(i, j int) bool { return b[i].Price < b[j].Price }
-func (b ByPrice) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
-
-// sortOrdersByPrice the caller function to sort orders
-func sortOrdersByPrice(o *orderSummary, reverse bool) {
-	if reverse {
-		sort.Sort(sort.Reverse(ByPrice(*o)))
-	} else {
-		sort.Sort(ByPrice(*o))
-	}
-}
-
-func (b *Base) findAmount(price float64, buy bool) (float64, orderSummary) {
-	orders := make(orderSummary, 0)
-	var amt float64
-
+func (b *Base) findAmount(priceTarget float64, buy bool) (*DeploymentAction, error) {
+	action := DeploymentAction{}
 	if buy {
+		if len(b.Asks) == 0 {
+			return nil, errNoLiquidity
+		}
+		action.ReferencePrice = b.Asks[0].Price
+		if action.ReferencePrice > priceTarget {
+			return nil, errUnableToHitPriceTarget
+		}
 		for x := range b.Asks {
-			if b.Asks[x].Price >= price {
-				amt += b.Asks[x].Price * b.Asks[x].Amount
-				orders = append(orders, Item{
-					Price:  b.Asks[x].Price,
-					Amount: b.Asks[x].Amount,
-				})
-				return amt, orders
+			if b.Asks[x].Price >= priceTarget {
+				action.TranchePositionPrice = b.Asks[x].Price
+				return &action, nil
 			}
-			orders = append(orders, Item{
-				Price:  b.Asks[x].Price,
-				Amount: b.Asks[x].Amount,
-			})
-			amt += b.Asks[x].Price * b.Asks[x].Amount
+			action.Orders = append(action.Orders, b.Asks[x])
+			action.QuoteAmount += b.Asks[x].Price * b.Asks[x].Amount
+			action.BaseAmount += b.Asks[x].Amount
 		}
-		return amt, orders
+		return nil, errNotEnoughLiquidity
 	}
 
+	if len(b.Bids) == 0 {
+		return nil, errNoLiquidity
+	}
+	action.ReferencePrice = b.Bids[0].Price
+	if action.ReferencePrice < priceTarget {
+		return nil, errUnableToHitPriceTarget
+	}
 	for x := range b.Bids {
-		if b.Bids[x].Price <= price {
-			amt += b.Bids[x].Amount
-			orders = append(orders, Item{
-				Price:  b.Bids[x].Price,
-				Amount: b.Bids[x].Amount,
-			})
-			break
+		if b.Bids[x].Price <= priceTarget {
+			action.TranchePositionPrice = b.Bids[x].Price
+			return &action, nil
 		}
-		orders = append(orders, Item{
-			Price:  b.Bids[x].Price,
-			Amount: b.Bids[x].Amount,
-		})
-		amt += b.Bids[x].Amount
+		action.Orders = append(action.Orders, b.Bids[x])
+		action.QuoteAmount += b.Bids[x].Price * b.Bids[x].Amount
+		action.BaseAmount += b.Bids[x].Amount
 	}
-	return amt, orders
+	return nil, errNotEnoughLiquidity
```

Comment:
It would be nice to get more information from these responses when liquidity is not enough - to remove guessing games. eg I want to slip to 1337 and I am told "can't be done", so I need to keep modifying values until I know what can be done. I can't even simulate being a whale 😭 

So in that case, is not enough liquidity an error? Or just a different kind of response (which contains an `errNotEnoughLiquidity`)?